src/master.c

```diff
@@ -303,75 +309,97 @@ bud_error_t bud_master_spawn_workers(bud_config_t* config) {
   bud_config_t* config;
   int i;
-  int j;
-  int r;
-  uv_process_options_t options;
-  char* aux_argv[] = {"--worker", NULL, NULL, NULL};
 
   config = worker->config;
-  ASSERT(config != NULL, "Worker config absent");
 
-  /* Config reload requested, this worker should not be restarted */
-  if (worker->state & kBudWorkerStateStale) {
-    bud_worker_t* start;
+  start = &worker[-worker->index];
+  worker->state |= kBudWorkerStateDead;
 
-    start = &worker[-worker->index];
-    worker->state |= kBudWorkerStateDead;
+  /* However check if all workers are gone, and if so - release the storage */
+  for (i = 0; i < config->worker_count; i++)
+    if (!(start[i].state & kBudWorkerStateDead))
+      return bud_ok();
 
-    /* However check if all workers are gone, and if so - release the storage */
-    for (i = 0; i < config->worker_count; i++)
-      if (!(start[i].state & kBudWorkerStateDead))
-        return bud_ok();
+  /* All gone :( Good bye! */
+  free(start);
+  return bud_ok();
+}
 
-    /* All gone :( Good bye! */
-    free(start);
-    return bud_ok();
-  }
 
-  memset(&options, 0, sizeof(options));
-  options.exit_cb = bud_master_respawn_worker;
-  options.file = config->exepath;
-  options.stdio_count = 3;
-  options.stdio = calloc(options.stdio_count, sizeof(*options.stdio));
+bud_error_t bud_master_get_spawn_args(bud_config_t* config, char*** out) {
+  char** args;
+  char* aux_argv[] = {"--worker", NULL, NULL, NULL};
+  int i;
+  int j;
 
-  /* config was piped to master's stdin, now inline it for the workers */
-  if (config->piped) {
-    aux_argv[1] = "-i";
-    aux_argv[2] = config->path;
-  }
+  args = calloc(config->argc + ARRAY_SIZE(aux_argv), sizeof(*args));
 
-  options.args = calloc(config->argc + ARRAY_SIZE(aux_argv),
-                        sizeof(*options.args));
+  if (args == NULL)
+    return bud_error_str(kBudErrNoMem, "options.args");
 
-  if (options.stdio == NULL || options.args == NULL) {
-    err = bud_error(kBudErrNoMem);
-    goto done;
+  /* config was piped to master's stdin, now inline it for the workers */
+  if (config->piped) {
+    aux_argv[1] = "-c";
+    aux_argv[2] = "!config";
   }
 
   /* Cases:
    * if piped_index <= 0:
    *    args = { config.argv, "--worker" }
    * otherwise:
-   *    args = { config.argv, "--worker", "-i", <config> }
+   *    args = { config.argv, "--worker", "-p", "!config" }
    */
   if (config->piped_index <= 0) {
     for (j = 0; j < config->argc; j++)
-      options.args[j] = config->argv[j];
+      args[j] = config->argv[j];
   } else {
     /* Goal is to skip piped_index, thus excluding --piped-config argument: */
     for (i = 0, j = 0; i < config->piped_index; i++, j++)
-      options.args[j] = config->argv[i];
+      args[j] = config->argv[i];
 
     for (i = config->piped_index + 1; i < config->argc; i++, j++)
-      options.args[j] = config->argv[i];
+      args[j] = config->argv[i];
+  }
+
+  for (i = 0; i < (int) ARRAY_SIZE(aux_argv); i++, j++)
```

Comment:
Might be okay for the case where ARRAY_SIZE(aux_argv) is small, but we are doing unnecessary work here.